ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/AddressIndexedSignerProvider.java

```diff
@@ -12,66 +12,41 @@
  */
 package tech.pegasys.ethsigner.core;
 
-import tech.pegasys.signers.secp256k1.EthPublicKeyUtils;
 import tech.pegasys.signers.secp256k1.api.Signer;
 import tech.pegasys.signers.secp256k1.api.SignerProvider;
 
 import java.security.interfaces.ECPublicKey;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
 
-import org.apache.tuweni.bytes.Bytes;
-import org.web3j.crypto.Keys;
+import com.google.common.annotations.VisibleForTesting;
 
 public class AddressIndexedSignerProvider {
-
-  // Provides a signer based on its Public Key
   private final SignerProvider signerProvider;
 
-  // String is an 0x prefixed hex string of the ethereum address corresponding to the public
-  // key.
-  private final Map<String, ECPublicKey> addressToPublicKeyMap;
-
-  public AddressIndexedSignerProvider(
-      final SignerProvider signerProvider, final Map<String, ECPublicKey> addressToPublicKeyMap) {
+  public AddressIndexedSignerProvider(final SignerProvider signerProvider) {
     this.signerProvider = signerProvider;
-    this.addressToPublicKeyMap = addressToPublicKeyMap;
   }
 
   public static AddressIndexedSignerProvider create(final SignerProvider signerProvider) {
-    final Map<String, ECPublicKey> addrToPubKeyMap = new HashMap<>();
-
-    signerProvider
-        .availablePublicKeys()
-        .forEach(
-            pubKey -> {
-              final String address =
-                  "0x"
-                      + Keys.getAddress(
-                              Bytes.wrap(EthPublicKeyUtils.toByteArray(pubKey)).toHexString())
-                          .toLowerCase();
-              addrToPubKeyMap.put(address, pubKey);
-            });
-
-    return new AddressIndexedSignerProvider(signerProvider, addrToPubKeyMap);
+    return new AddressIndexedSignerProvider(signerProvider);
   }
 
   /* Gets a signer from its address, NOTE address MUST have 0x hex prefix */
   public Optional<Signer> getSigner(final String address) {
-    final ECPublicKey publicKey = addressToPublicKeyMap.get(address.toLowerCase());
-    if (publicKey == null) {
-      return Optional.empty();
-    }
-    return signerProvider.getSigner(publicKey);
+    return signerProvider.getSigner(new Eth1AddressSignerIdentifier(address));
   }
 
+  @VisibleForTesting
   public Set<String> availableAddresses() {
-    return addressToPublicKeyMap.keySet();
+    return availablePublicKeys().stream()
```

Comment:
Don't think this should be part of this class. The public keys can easily be converted to addresses if needed. And looks like where it is used doesn't even care about addresses just wants at least one key to exist.

Reply:
yes, you are right ... let me remove this method.